src/index.ts

```diff
@@ -46,12 +48,38 @@ const shouldInclude = (name: string, config: HardhatDeployOptions): boolean => {
     }
 };
 
+const shouldIncludeFile = (
+    fileName: string,
+    config: HardhatDeployOptions
+): boolean => {
+    // Extract the network name from the file name (assumes format "networkName.json")
+    const networkName = fileName.split(".")[0];
```

Comment:
Use `path.basename(fileName, ".json")`
https://nodejs.org/api/path.html#pathbasenamepath-suffix

Reply:
Adjusted. Thanks for the note.